internal/frontend/token/token.go

```diff
@@ -91,33 +91,14 @@ func (T *Token) UintValue() (uint64, error) {
 	return strconv.ParseUint(string(T.Lit), 10, 64)
 }
 
+var (
+	sdtPlaceholders = regexp.MustCompile(`\$(\d+)`)
+	sdtReplacement  = []byte("X[$1]")
+)
+
```

Comment:
I think a small comment here explaining the functionality of gocc would be helpful in future. Something along the lines of -
```
// SDTVal converts a handwritten action expression into an equivalent expression
// usable during runtime.
// For e.g. the expression `<< $0.(int64) + $2.(int64), nil >>` is converted to
// `X[0].(int64) + X[2].(int64), nil`. `X` usually represents a slice of
// attributes in the code generated by gocc.
```

Reply:
there was no such explanation in the original code, so I do did not add it :)

are you sure any explanation is required here? this code is completely self-explaining FMPOV